src/Microsoft.Diagnostics.Runtime/src/Linux/LinuxLiveDataReader.cs

```diff
@@ -0,0 +1,517 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+#if !NET45
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Diagnostics.Runtime;
+
+namespace Microsoft.Diagnostics.Runtime.Linux
+{
+    /// <summary>
+    /// A data reader targets a Linux process, implemented by reading /proc/<pid>/maps 
+    /// and /proc/<pid>/mem files. The process must have READ permission to the above 2
+    /// files. 
+    ///   1. The current process can run as root.
+    ///   2. If executed from within a Docker container, the best way is to use "ptrace 
+    ///      attach" to obtain the permission. 
+    ///        - the container should be started with "--cap-add=SYS_PTRACE" or equivalent. 
+    ///        - the process must call the following before constructing the data reader.
+    ///             if (ptrace(PTRACE_ATTACH, targetProcessId, NULL, NULL) != 0) { fail }
+    ///             wait(NULL);
+    /// </summary>
+    internal class LinuxLiveDataReader : IDataReader2
+    {
+        private List<MemoryMapEntry> _memoryMapEntries;
+        private FileStream _memoryStream;
+        private List<uint> _threadIDs = new List<uint>();
+
+        public LinuxLiveDataReader(uint processId)
+        {
+            this.ProcessId = processId;
+            _memoryMapEntries = this.LoadMemoryMap();
+        }
+
+        public uint ProcessId { get; private set; }
+
+        public bool IsMinidump { get { return false; } }
+
+        public void Close()
+        {
+            _memoryStream?.Dispose();
+            _memoryStream = null;
+        }
+
+        public void Flush()
+        {
```

Comment:
This should be:

    _threadIDs.Clear();
    _memoryMapEntries = null;
    _memoryStream = LoadMemoryMap();

Reply:
I see. changed to:
            _threadIDs.Clear();
            _memoryStream?.Dispose();
            _memoryStream = null;
            _memoryMapEntries = this.LoadMemoryMap();